internal/own/codeowners/proto/find_owners.go

```
@@ -43,19 +45,28 @@ type anySubPath struct{}
 func (p anySubPath) String() string      { return "**" }
 func (p anySubPath) Match(_ string) bool { return true }
 
-// exactMatch is indicated by an exact name of directory or a file within
-// the glob pattern, and matches that exact part of the path only.
-type exactMatch string
-
-func (p exactMatch) String() string         { return string(p) }
-func (p exactMatch) Match(part string) bool { return string(p) == part }
+// anyMatch is a pattern that matches any path part.
+var anyMatch patternPart = makeAsteriskPattern("*")
 
-// anyMatch is indicated by * in a glob pattern, and matches any single file
-// or directory on the path.
-type anyMatch struct{}
+// asteriskPattern is a pattern that may contain * glob wildcard.
+type asteriskPattern struct {
+	glob     string
+	compiled *regexp.Regexp
+}
 
-func (p anyMatch) String() string      { return "*" }
-func (p anyMatch) Match(_ string) bool { return true }
+func makeAsteriskPattern(pattern string) asteriskPattern {
+	quoted := regexp.QuoteMeta(pattern)
+	regular := strings.ReplaceAll(quoted, `\*`, `.*`)
+	compiled, err := regexp.Compile("^" + regular + "$")
+	if err != nil {
+		return asteriskPattern{}
+	}
```

should this panic? doesn't seem like it should ever error, but swallowing also feels bad :)